src/Agent.Listener/Configuration/AutoLogonConfigurationManager.cs

```diff
@@ -0,0 +1,283 @@
+#if OS_WINDOWS
+using Microsoft.VisualStudio.Services.Agent.Util;
+using Microsoft.TeamFoundation.DistributedTask.WebApi;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Security.Principal;
+using System.Threading;
+
+namespace Microsoft.VisualStudio.Services.Agent.Listener.Configuration
+{
+    [ServiceLocator(Default = typeof(AutoLogonConfigurationManager))]
+    public interface IAutoLogonConfigurationManager : IAgentService
+    {
+        void Configure(CommandSettings command);
+        void UnConfigure();
+        bool RestartNeeded();
```

Comment:
1) why to ad another store when registry is already there.
2) In case of different user configuring the autologon, whether the restart question was asked to the user or not.

---
In reply to: [118063920](https://github.com/Microsoft/vsts-agent/pull/985#discussion_r118063920) [](ancestors = 118063920)